### PR TITLE
lyra: modernize

### PR DESCRIPTION
**Pull request by prince-chrismc**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by prince-chrismc** [comment]
👋 @SpaceIm @ericLemanissier @toge i'd really appreciate if you can double check me 🙏 
trying to get back into the swing of things ❤️

File: recipes/lyra/all/conanfile.py [hunk 1]
```diff
@@ -30,9 +31,13 @@ def source(self):
...
     def package(self):
         self.copy("LICENSE.txt", dst="licenses", src=self._source_subfolder)
-        self.copy("*.h*", dst="include", src=self._source_subfolder+"/include")
+        self.copy("*.h*", dst="include", src=os.path.join(self._source_subfolder, "include"))
 
     def package_info(self):
+        self.cpp_info.set_property("cmake_file_name", "lyra")
+        self.cpp_info.set_property("cmake_target_name", "bfg::lyra")
+
+        # TODO: to remove in conan v2 once cmake_find_package* generators removed
```
Comment (SpaceIm) [inline recipes/lyra/all/conanfile.py:40]:
```suggestion
        # TODO: to remove in conan v2 once cmake_find_package* generators removed
        self.cpp_info.components["_lyra"].set_property("cmake_target_name", "bfg::lyra")
```

Comment (prince-chrismc) [reply recipes/lyra/all/conanfile.py:40]: 🤔 Why is it better to add a component? AFAIK this project only has one part for I thought the "global space" was better.

Thanks in advance for helping me along

Comment (SpaceIm) [reply recipes/lyra/all/conanfile.py:40]: there is a component due to cmake_find_package, it avoids to declare un unwanted target.

**Review by toge — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**